tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino

```diff
@@ -0,0 +1,385 @@
+  x += 180;  // no negative numbers
+  QMC5883L.HG = x;
+
+  // calculate scalar magnetic induction
+  QMC5883L.scalar = sqrt((QMC5883L.MX * QMC5883L.MX) + (QMC5883L.MY * QMC5883L.MY) + (QMC5883L.MZ * QMC5883L.MZ));
+ 
+   // get temperature
+  //if (readRegister(QMC5883L_TEMP_LSB, 2) != 2)   return; // read error
+  if (I2cValidReadS16(&QMC5883L.temp, QMC5883L.i2c_address, QMC5883L_TEMP_LSB) == false)  return; // read error, select LSB register
+  //t = Wire.read() | (Wire.read() << 8);
+  QMC5883L.temp = (QMC5883L.temp / 100) + USE_QMC5883L_Temp;
+}
+
+/*********************************************************************************************\
+ * Presentation
+\*********************************************************************************************/
+
+#ifdef USE_WEBSERVER
+const char HTTP_SNS_QMC5883L[] PROGMEM =
+    "{s}QMC5883L " D_MX "{m}%d " D_UNIT_MICROTESLA "{e}"                        // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+    "{s}QMC5883L " D_MY "{m}%d " D_UNIT_MICROTESLA "{e}"                        // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+    "{s}QMC5883L " D_MZ "{m}%d " D_UNIT_MICROTESLA "{e}"                        // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+    "{s}QMC5883L " D_MAGNETICFLD "{m}%d " D_UNIT_MICROTESLA "{e}"               // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+    "{s}QMC5883L " D_HG "{m}%d " D_UNIT_DEGREE "{e}"                            // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+    "{s}QMC5883L " D_TEMPERATURE "{m}%d " D_UNIT_DEGREE D_UNIT_CELSIUS "{e}"; // {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+
+const char HTTP_SNS_QMC5883L_ERROR[] PROGMEM =
+    "{s}QMC5883L {m} %s {e}";
+#endif
+
+
+void QMC5883L_Show(uint8_t json)
+{
+  if (json) 
+  {
+    if (QMC5883L.ready != true)
+    {
+      AddLog(LOG_LEVEL_INFO, PSTR("QMC5883L: " D_ERROR " %x" ), QMC5883L.status);
+      return;
+    }
+    else 
+    {
+      ResponseAppend_P(PSTR(",\"QMC5883L\":{\"" D_JSON_MX "\":%d,\"" D_JSON_MY "\":%d,\"" D_JSON_MZ "\":%d,\"" D_JSON_MAGNETICFLD "\":%u,\"" D_JSON_HEADING "\":%d,\"" D_JSON_TEMPERATURE "\":%d}"), QMC5883L.MX, QMC5883L.MY, QMC5883L.MZ, QMC5883L.scalar, QMC5883L.HG, QMC5883L.temp);
+    }
+  }
+#ifdef USE_WEBSERVER
+  else 
+  {
+    if (QMC5883L.ready != true)
+    {
+        WSContentSend_PD(HTTP_SNS_QMC5883L_ERROR, D_ERROR);    
+    }
+    else
+    {
+        WSContentSend_PD(HTTP_SNS_QMC5883L, QMC5883L.MX, QMC5883L.MY, QMC5883L.MZ, QMC5883L.scalar, QMC5883L.HG, QMC5883L.temp);
+    }    
+  }
+#endif
+}
+
+/*********************************************************************************************\
+ * Interface
+\*********************************************************************************************/
+ 
+bool Xsns33(byte function)
+{
+  if (!I2cEnabled(XI2C_71))
+  {
+    return false;
+  }
+
+  if (FUNC_INIT == function) {
+    QMC5883L_Init();
+  }
+  else 
+  if (QMC5883L.ready == true) 
```

Comment:
```
if (QMC5883L)
```
and please put the if on the same line as the else

Reply:
Overall this also mean you son't need anymore the `ready` and the `i2c_address` of the structure
Even more RAM saving 😄